bin/brew

```diff
@@ -44,6 +44,14 @@ fi
 
 HOMEBREW_LIBRARY="$HOMEBREW_REPOSITORY/Library"
 
+for VAR in EDITOR PATH
+do
+  VAR_NEW="HOMEBREW_${VAR}"
+  # TODO: find a better solution than this.
+  env | grep -q "$VAR_NEW" && continue
```

Comment:
```bash
[[ -z ${!VAR_NEW+x} ]] || continue
```

`!` does indirect variable expansion; `+x` tests for whether the variable is unset, and substitutes `x` iff it is, so the `-z` condition is true iff `$VAR_NEW` is unset.